File: 0135-candy/0135-candy.java
class Solution {
    public int candy(int[] nums) {
        int ans = 0;
        int last = -1;
        int[] left = new int[nums.length];
        int[] right = new int[nums.length];
        Arrays.fill(left, 1);
        Arrays.fill(right, 1);

        for(int i=1;i<nums.length;i++){
          if(nums[i] > nums[i-1]){
            left[i] = left[i-1]+1;
          }
          else{
            left[i]= 1;
          }
        }
        for(int i=nums.length-2;i>=0;i--){
          if(nums[i] > nums[i+1]){
            right[i] = right[i+1]+1;
          }
          else{
           right[i]= 1;
          }
        }

        for(int i=0;i<nums.length;i++){
        ans += Math.max(left[i],right[i]);
        }

        return ans;
    }
}

File: 0135-candy/0135-candy.cpp
class Solution {
public:
    int candy(vector<int>& ratings) {
        vector<int>left(ratings.size(),1);
         vector<int>right(ratings.size(),1); int sum=0;
        for(int i=1;i<ratings.size();i++){
             if(ratings[i]>ratings[i-1]){
                left[i] = left[i-1]+1;
             }else left[i] = 1;
        }

         for(int i=ratings.size()-2;i>=0;i--){
             if(ratings[i]>ratings[i+1]){
                right[i] = right[i+1]+1;
             }else right[i] = 1;
        }

        for(int i=0;i<ratings.size();i++){
           sum += max(left[i],right[i]);
        }
        return sum;




        
    }
};